### PR TITLE
Rx View History Page

### LINKED ISSUE #2583: [Rx] RxHistoryPage component
Blocks by #2571 

**Contains / blocked by:**
 
- GrowableTable component (#2584)
- Pagination component (#2585)

### PR DESCRIPTION
**Pull request by U-DON**
To close #2583, #2584.

Still have some things to sort out (no pun intended...) with sorting and pagination. Need to be able to sort in reverse order, indicate which parameter the history is being sorted by, etc. Right now, clicking on each table header will sort by that field in ascending order.

Just putting up the WIP since I refactored the prescriptions actions and reducers a bit and didn't want to blindside anyone who might be touching the same files. Still figuring out how I want to consolidate the actions.

### CODE REVIEW

File: src/js/rx/actions/prescriptions.js [hunk 1]
```diff
@@ -21,7 +21,31 @@ export function loadData(id) {
     };
   }
 
-  return dispatch => fetch('/rx-api/prescriptions')
+  return dispatch => dispatch({ type: 'LOAD_PRESCRIPTION_FAILURE' });
+}
+
+export function loadPrescriptions(options) {
+  let uri = '/rx-api/prescriptions';
```
Comment (webinista) [inline src/js/rx/actions/prescriptions.js]: Late incoming change! I think this should now be `/api/v0/prescriptions`.

Comment (U-DON) [inline src/js/rx/actions/prescriptions.js]: Oh, I think that change would be in our local `config.proxy.js` files (`path: '/api/v0'`). We're using `/rx-api/[rest_of_uri]` for development, I think. Don't know if we keep it this way for production, but @alexose might know.

File: src/js/rx/containers/History.jsx [hunk 2]
```diff
@@ -1,26 +1,69 @@
...
+    super(props);
+    this.handleSort = this.handleSort.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.dispatch(loadPrescriptions());
+  }
+
+  handleSort(param) {
+    // console.log('Sorting happens here...');
```
Comment (webinista) [inline src/js/rx/containers/History.jsx]: Can remove this.

**Comment by U-DON** [comment]
Ready for review. Note that testing this has been a little frustrating since the `GET /prescriptions` request can sometimes fail. But when it did work, I was able to get the table to sort with all the fields in both orders.

I was hoping to get the history view's state into its own reducer (for its own `sort` state), but I didn't have time to figure out how to dispatch actions for different reducers from the prescriptions action creator (or whether that's kosher).

I wasn't able to get to pagination, but perhaps someone else can pick it up. I was hoping to put that into the reducer for the history view along with the sort value.

Ultimately, I would like to get both the sort and pagination from the current URL as query params and have it working with react-redux-router.

**Comment by U-DON** [comment]
Few things still being worked on (should be in today):
1. Making the table collapse into a list on mobile.
2. Adding the fill date field and sort option.
3. Turning some fields into links (prescription name to detail page, status to glossary).

**Comment by U-DON** [comment]
Also added sorting in mobile view. Due to the unreliability of `GET /prescriptions` requests, the dropdown menu behavior can be a little frustrating. When it fails, the selection doesn't change, and there's no indication that the request failed.

It's a similar issue of timing whenever we request data from the API, and I think we can add transitions, loading animations, and messaging to better communicate when we are fetching data or when the request fails. Anyway, that is a separate bundle of work. I think I've reached a good stopping point for this PR for review. Pagination will be done in a separate branch.

## Mocks
Desktop: https://marvelapp.com/1h10heg#14227014
Mobile: https://marvelapp.com/iaa9b9#14487574

## Desktop Preview
screenshot from 2016-09-06 17 31 17

## Mobile Preview
screenshot from 2016-09-06 17 31 34

**Comment by webinista** [comment]
LGTM! Will merge when tests pass.

**Pull request merged**